api/src/main/java/jakarta/data/metamodel/TextAttribute.java

```diff
@@ -40,4 +43,84 @@ public interface TextAttribute<T> extends SortableAttribute<T> {
      */
     Sort<T> descIgnoreCase();
 
+    /**
+     * Creates a `LIKE` restriction for an exact match on the specified text.
+     *
+     * @param text the text to match exactly.
+     * @return a Restriction representing a case-sensitive `LIKE` condition.
```

Comment:
Jakarta Data has never guaranteed case sensitive comparisons. Some databases either are or have modes in which they are case insensitive, and the Jakarta Data provider has no control over that.  So we very careful wrote the documentation to say only there are ways to specifically request case insensitive comparisons, not that we can ever ensure that you will get case sensitive.

```suggestion
     * Creates a `LIKE` restriction for a match on the specified text.
     *
     * @param text the text to match.
     * @return a Restriction representing a `LIKE` condition.
```